Add an example for an image in annotation

lpoujade opened the pull request:
Follow up of https://github.com/IIIF/cookbook-recipes/issues/377 , opened a pull request to generate the preview, which is [accessible here](https://preview.iiif.io/cookbook/377-image-in-annotation/recipe/377-image-in-annotation/)

It probably miss something cause I don't see my recipe [on this page](https://preview.iiif.io/cookbook/377-image-in-annotation/)

Also, I don't know what to put in the Restrictions part

thehabes commented:
Your preview did build!  It is at https://preview.iiif.io/cookbook/377-image-in-annotation/recipe/377-image-in-annotation/

glenrobson commented:
Sorry @lpoujade I forgot to paste out comments into this pull request but we had a look at your recipe in the call last week and noted the following:

# Use case
 * Expand the use case with a particle example of how you want to do this e.g. I might have a picture of a person named in the manuscript or something else…
 * It might be useful to show (with a small screenshot or partial screenshot) what you expect this to look like in a viewer. What do you do with large or small images. 

# Implementation note 
 * It would be useful to talk about how this recipe differs from the layers recipe?  A different example may help as two versions of the same images could be done as a layer. Maybe a text based object and a related image might be clearer.  
 * Add to implementation notes other IIIF ways to achieve similar aims

# Restrictions 
 * We couldn't come up with any restrictions. 

# Example
 * Group likes the way you've done it with two bodies   
 * Viewers only include the one that works and note there is an issue with Annona which is fixed in #394  
 * Mirador also dosen’t show the image part of the annotation so not supported. 

Also could you add a link to this recipe in the following file: https://github.com/IIIF/cookbook-recipes/blob/377-image-in-annotation/index.md and also in the [links.md](https://github.com/IIIF/cookbook-recipes/blob/377-image-in-annotation/_includes/links.md) (note the number is the issue id so 0037 in this case).

lpoujade commented:
Thank you @glenrobson ! I had see the comments on the google doc but didn't had enough time to update my pr. I'll try to do it before the next cookbook call

About the layer recipe, I assume you are talking about [this one](https://iiif.io/api/cookbook/recipe/0326-annotating-image-layer/) ?

lpoujade commented:
Hi, I updated my PR but got some questions:
- about how to render an image in a viewer, I was thinking to something like [this](https://user-images.githubusercontent.com/12305084/118143390-47def000-b3d9-11eb-8e92-92ee6de5e837.png) (taken from [this PR for Mirador](https://github.com/ProjectMirador/mirador/pull/3435)), also [the way Annona render it](https://ncsu-libraries.github.io/annona/tools/#/display?url=https%3A%2F%2Fpreview.iiif.io%2Fcookbook%2F377-image-in-annotation%2Frecipe%2F377-image-in-annotation%2Fmanifest_at_id.json&viewtype=iiif-storyboard&manifesturl=&settings=%7B%22fullpage%22%3Atrue%7D) is really nice. But I don't have a clear idea in mind, I was thinking this could be at the discretion of the viewer. I could add thoses screenshots to the recipe, but i'm not sure if you wanted them here or in the recipe ?
- about the layer recipe I don't really see the similarity, my proposal is only about associating a picture with a small part of the main data, not an alternate representation of this data. I added some example use cases, does is clear the confusion ?

dnoneill commented:
This works in Annona for any id field now. Also the caption and attribution fields are `description` and `attribution` in the annotation. I think I got this based on an in the wild example but I have fixed Annona so it should not create undefined values if they don't exist in the annotation. Thanks for adding this, I only had one example to work from before and it was v2.

lpoujade commented:
Cool! Glad it could help :)
I updated the PR

glenrobson commented on recipe/0377-image-in-annotation/index.md:
```diff
@@ -0,0 +1,48 @@
+
+## Implementation Notes
+
+Our supplementary image is not part of the Canvas content, thus it must not have the motivation `painting`, and so is to be placed in the `annotations` section of the manifest. Following the [W3C Web Annotation model](https://www.w3.org/TR/annotation-model/#external-web-resources), we have a simple image body referencing our image, and a text body to provides a comment/title about the image.
```
Could you change:

`annotations` section of the manifest

to:

`annotations` section of the Canvas.

As the annotations are associated with a Canvas

glenrobson commented on recipe/0377-image-in-annotation/index.md:
```diff
@@ -0,0 +1,48 @@
+* [Annotate specific images or layers][0326]
+  * in our case, we had only one image so we annotated the main canvas
+* [Annotation with a Non-Rectangular Polygon][0261]
+  * we could have drawn the contours of the foutain
```
Miss spelling of fountain

glenrobson commented on recipe/0377-image-in-annotation/index.md:
```diff
@@ -0,0 +1,48 @@
+
+## Example
+
+The main content is a photo of a square in Göttingen, which shows, among others things, a fountain. We wanted to show the lights on the foutain during the night, so we associated the part of the canvas containing the foutain with an Annotation consisting of a picture of the foutain at night and a text comment.
```
3 miss spellings of fountain

glenrobson commented on recipe/0377-image-in-annotation/index.md:
```diff
@@ -0,0 +1,48 @@
+
+Fixtures used in this manifest:
+- [main content](https://fixtures.iiif.io/info.html?file=/images/Glen/photos/gottingen.jpg)
+- [image in annotation](https://fixtures.iiif.io/info.html?file=/images/Glen/photos/fountain.jpg)
```
You can remove this part as readers will be able to see the images from the Manifest. If you do want to include it could you move it to the Example section.

glenrobson commented on recipe/0377-image-in-annotation/index.md:
```diff
@@ -0,0 +1,48 @@
+
+The main content is a photo of a square in Göttingen, which shows, among others things, a fountain. We wanted to show the lights on the foutain during the night, so we associated the part of the canvas containing the foutain with an Annotation consisting of a picture of the foutain at night and a text comment.
+
+> {% include manifest_links.html viewers="Annona" manifest="manifest.json" %}
```
Could you remove the

`> `

At the start of this line?

glenrobson commented:
The cookbook editors have reviewed your recipe and requested the following changes:

 * Could you add an 's' to annotations in the title: "Image in annotations" (this is to match the HTML in annotations recipe)
 * Patrick is going to leave notes on the pull request regarding a few small language issues in the use case
 * Could you remove the !!! marks from the Tagging with an external resource link in the related recipes. 
 * Could you remove the sub bullets in the related recipes and move the text next to the link so it looks similar to the related recipes in: https://iiif.io/api/cookbook/recipe/0019-html-in-annotations/ 

With the above changes the editors are happy for it to go to the TRC for final review. Would you be able to make the above changes by next Wednesday?

Thanks

lpoujade commented:
Yes I can update my PR next week 

Le 27 avril 2023 15:48:07 GMT+02:00, Glen Robson ***@***.***> a écrit :
>The cookbook editors have reviewed your recipe and requested the following changes:
>
> * Could you add an 's' to annotations in the title: "Image in annotations" (this is to match the HTML in annotations recipe)
> * Patrick is going to leave notes on the pull request regarding a few small language issues in the use case
> * Could you remove the !!! marks from the Tagging with an external resource link in the related recipes. 
> * Could you remove the sub bullets in the related recipes and move the text next to the link so it looks similar to the related recipes in: https://iiif.io/api/cookbook/recipe/0019-html-in-annotations/ 
>
>With the above changes the editors are happy for it to go to the TRC for final review. Would you be able to make the above changes by next Wednesday?
>
>Thanks
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/IIIF/cookbook-recipes/pull/392#issuecomment-1525727553
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
